### ISSUE TITLE
[V16] Issues with Spotlights

### ISSUE THREAD

**Issue by AshCraft1337** [Roblox/future-is-bright#145]
![image](https://user-images.githubusercontent.com/30395035/50541626-b2c85d00-0b77-11e9-8021-d5215d807183.png)

Spotlights won't cast shadows ^

![image](https://user-images.githubusercontent.com/30395035/50541630-c247a600-0b77-11e9-8be5-2103680d2ee1.png)

Spotlights produce this glitch when against a wall ^

![image](https://user-images.githubusercontent.com/30395035/50541650-f91dbc00-0b77-11e9-83c4-91216051e99e.png)

Spotlight at angle 180 ^

**Comment by Rocky28447** [comment]
Yes, having this issue as well. Seems to occur when the brightness and angle are too high.

![image](https://user-images.githubusercontent.com/26193486/50573523-70836380-0da3-11e9-9edd-c5cc4558c717.png)
![image](https://user-images.githubusercontent.com/26193486/50573527-78db9e80-0da3-11e9-94d2-2d22115dc9cc.png)

**Comment by usahappy** [comment]
I'm having the same issue with spot lights not casting shadows. I've actually only gotten point lights to cast shadows, but surface lights haven't been working for a couple versions now. 
From left to right, surface light, point light, and spot light:
<img width="728" alt="screen shot 2019-01-01 at 11 57 35 pm" src="https://user-images.githubusercontent.com/28405315/50580843-cdbdfa00-0e21-11e9-969d-d3042e4631c9.png">

**Comment by Hifiveghost5** [comment]
me to
![robloxscreenshot20190109_175610074](https://user-images.githubusercontent.com/43627120/50941214-cc4f8b80-1438-11e9-8967-22993232f199.png)

**Comment by UncleGPU** [comment]
SpotLights and Surface lights don't cast shadows in v16. Only PointLights cast shadows
The spotlight bug occurs when you have the angle set to 180. Reduce it to get rid of the effect
